### ISSUE TITLE
[BUG] Salt-Minion servie status is failed after the successful installation of Salt-minion on SDDC Manager appliance running on Photon OS 4

### ISSUE THREAD

**Issue by MadhusudanMI-PSH** [saltstack/salt#65568]
**Description**
Installed Salt-minion(3006.3) on SDDC Manager appliance running on Photon OS 4. 

After the installation checking the status of the service fails with the issue shown below: 

**Steps to Reproduce the behavior**
Install salt-minion on SDDC Manager appliance running on Photon OS 4.
Enable and start salt-minion service.
Check the status of the salt-minion service to see the error.

**Expected behavior**
Expected it to be running.

**Screenshots**
```
root@sddc-manager [ /home/vcf ]# sudo systemctl enable salt-minion && sudo systemctl start salt-minion
Created symlink /etc/systemd/system/multi-user.target.wants/salt-minion.service → /usr/lib/systemd/system/salt-minion.service.
root@sddc-manager [ /home/vcf ]# sudo systemctl status salt-minion
● salt-minion.service - The Salt Minion
   Loaded: loaded (/usr/lib/systemd/system/salt-minion.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2023-11-15 10:22:37 UTC; 22s ago
    Docs: man:salt-minion(1)
       file:///usr/share/doc/salt/html/contents.html
       https://docs.saltproject.io/en/latest/contents.html
  Process: 18164 ExecStart=/usr/bin/salt-minion (code=exited, status=64)
  Main PID: 18164 (code=exited, status=64)
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:   import salt.channel.client
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/channel/client.py", line 13, in <module>
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:   import salt.crypt
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/crypt.py", line 1422, in <module>
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:   class Crypticle:
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:  File "/opt/saltstack/salt/lib/python3.10/site-packages/salt/crypt.py", line 1432, in Crypticle
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]:   SIG_SIZE = hashlib.sha256().digest_size
Nov 15 10:22:32 sddc-manager.vrack.vsphere.local salt-minion[18171]: ValueError: [digital envelope routines] unsupported
Nov 15 10:22:37 sddc-manager.vrack.vsphere.local systemd[1]: salt-minion.service: Main process exited, code=exited, status=64/USAGE
Nov 15 10:22:37 sddc-manager.vrack.vsphere.local systemd[1]: salt-minion.service: Failed with result 'exit-code'.
root@sddc-manager [ /home/vcf ]# timed out waiting for input: auto-logout
```

**Comment by s0undt3ch** [comment]
There's no mention that FIPS is enabled, though, the traceback suggests that's the case, can you confirm?

**Comment by s0undt3ch** [comment]
Although, `sha256` should be fine under FIPS ...

**Comment by dmurphy18** [comment]
Also interested in how the salt-minion was installed on Photon 4, that is, all of the steps taken to install the salt-minion.  Reason, from meeting last Thursday, the instructions used initially had the team creating their own salt-minion.service file

**Comment by dwoz** [comment]
This was fixed in 3006.4.
